recipes/docker_cgroup.rb

```diff
@@ -0,0 +1,69 @@
+docker_cgroup_cpu_cfs_quota_us = -1
+docker_cgroup_memory_hard_limit = 9223372036854771712
+docker_cgroup_memory_soft_limit = 9223372036854771712
+if node['hops']['docker']['cgroup']['enabled'].eql?("true")
+  docker_cgroup_memory_hard_limit = "#{node['hops']['docker']['cgroup']['memory']['hard-limit']}"
```

Comment:
For a "dynamic" way to set memory based on available memory in the host, do this:

create in attributes.default

node['hops']['docker']['cgroup']['memory']['hard-limit-default'] = 6
node['hops']['docker']['cgroup']['memory']['hard-limit'] = node['hops']['docker']['cgroup']['memory']['hard-limit-default']

node['hops']['docker']['cgroup']['memory']['soft-limit-default'] = 4
node['hops']['docker']['cgroup']['memory']['soft-limit'] = node['hops']['docker']['cgroup']['memory']['soft-limit-default']



Then in docker_cgroup.rb

available_memory_in_host_gbs = node.memory.total[/\d*/].to_i
default_hard_memory = node['hops']['docker']['cgroup']['memory']['hard-limit']
default_soft_memory = node['hops']['docker']['cgroup']['memory']['soft-limit']

available_memory_in_host_gbs = available_memory_in_host_gbs - default_hard_memory
excess_memory = available_memory_in_host_gbs - 30
excess_memory = excess_memory / 4

  # if the user overrides the hard-limit, then we should use their new hard-limit, and not add excess_memory
if node['hops']['docker']['cgroup']['memory']['hard-limit-default'] != node['hops']['docker']['cgroup']['memory']['hard-limit']
  excess_memory = 0
end

if node['hops']['docker']['cgroup']['memory']['soft-limit-default'] != node['hops']['docker']['cgroup']['memory']['soft-limit']
  excess_memory = 0
end

node.override['hops']['docker']['cgroup']['memory']['hard-limit'] = default_hard_memory + excess_memory
node.override['hops']['docker']['cgroup']['memory']['soft-limit'] = default_soft_memory + excess_memory